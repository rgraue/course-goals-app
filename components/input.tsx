import { goalInputStyle } from "@styles";
import { TextInput, View, Button } from "react-native";
import { useState } from "react";

export const GoalInput = ({callback}) => {
  const [enteredGoal, setEnteredGoal] = useState('');

  function goalInputHandler (enteredGoal: string) {
    setEnteredGoal(enteredGoal);
  }

  function addGoalHandler () {
    callback(enteredGoal);
    goalInputHandler('')
  }

  return (
    <View style={goalInputStyle.container}>
      <TextInput
        placeholder="Your course goal"
        placeholderTextColor={"#cccccc"}
        style={goalInputStyle.textEntry}
        onChangeText={goalInputHandler}
        value={enteredGoal}
      />
      <Button 
        title="Add Goal"
        color={"black"}
        onPress={addGoalHandler}
      />
    </View>
  )
}

