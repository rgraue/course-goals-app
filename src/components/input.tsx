import { inputStyle } from '@styles';
import { TextInput, View, Button } from 'react-native';
import { useState } from 'react';

export const Input = ({ handlerEnteredText }) => {
  const [enteredText, setEnteredText] = useState('');

  function textInputHandler(enteredGoal: string) {
    setEnteredText(enteredGoal);
  }

  const onPressHandler = () => {
    handlerEnteredText(enteredText);
  };

  return (
    <View style={inputStyle.container}>
      <TextInput
        placeholder="Enter Goal"
        placeholderTextColor={'#cccccc'}
        style={inputStyle.textInput}
        onChangeText={textInputHandler}
        value={enteredText}
      />
      <Button
        title="Add Goal"
        color={'#A396B0'}
        accessibilityLabel="Text Btn"
        onPress={onPressHandler}
      />
    </View>
  );
};
