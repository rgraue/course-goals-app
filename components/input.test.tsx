import React from 'react';
import { create } from 'react-test-renderer';
import { GoalInput } from './input';

function mockCallback() {
  console.log('hello world');
}

test('GoalInput Component', () => {
  const tree = create(<GoalInput callback={mockCallback} />);

  expect(tree.toJSON()).toMatchSnapshot();
});
