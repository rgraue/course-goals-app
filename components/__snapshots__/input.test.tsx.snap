// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test lol GoalInput Component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": "5%",
      "marginLeft": "5%",
      "marginRight": "5%",
      "marginTop": "10%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Your course goal"
    placeholderTextColor="#cccccc"
    style={
      Object {
        "borderBottomColor": "#cccccc",
        "borderBottomWidth": 1,
        "flex": 1,
        "marginEnd": "10%",
        "textAlign": "center",
        "width": "60%",
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "black",
            },
          ]
        }
      >
        Add Goal
      </Text>
    </View>
  </View>
</View>
`;
